### DISCUSSION TITLE
OIDC Attribute mapping

### DISCUSSION THREAD

**Discussion by iali9906** [outline/outline#7027] [Self Hosting]
hello,

I have configured the OIDC settings to use keycloack with the default settings, but after restricting access by domain, some users can't log in because they have a different domain.

is it possible to make outline during authentication go to read the mailAlternateAddress attribute instead of the mail attirbute?

possibly if i were to do a mapper on keycloack, how should it be?

Thanks

**Comment by tommoor** [comment]
I don't see a reason the login couldn't consider both attributes if someone wanted to make that change

**Reply by iali9906** [reply]
I can do this with the docker image via the env file?

**Reply by tommoor** [reply]
No, it would need changes to the code to read that attribute

**Reply by iali9906** [reply]
ok if i make the change to the code how can i implement it?
in the sense that I don't want to do the installation via source, but run it through docker anyway, how can I recreate the image after I make the change to the code? I saw that there are several files inherent in docker.

**Reply by Limezy** [reply]
I guess @tommoor is proposing you to do a PR that once finalized and tested could be merged into future versions of Outline

**Reply by iali9906** [reply]
yes it is clear, but before doing the PR, I need to see the code and understand which file is used for the authentication to be modified, I also wanted to customize the one I currently have in prod to modify the email templates.
